Wrappers/Python/cil/optimisation/algorithms/ADMM.py

```diff
@@ -111,21 +111,18 @@ def update(self):
         
         self.tmp_adj *= -(self.tau/self.sigma)
         self.x += self.tmp_adj     
+      
         # apply proximal of f        
-        tmp = self.f.proximal(self.x, self.tau)
-        self.x.fill(tmp)
-        del tmp
-        
-        self.operator.direct(self.x, out = self.tmp_dir)  
-        self.u += self.tmp_dir
+        self.f.proximal(self.x, self.tau, out=self.x)
+        self.operator.direct(self.x, out = self.tmp_dir)
         
         # apply proximal of g   
-        self.g.proximal(self.u, self.sigma, out = self.z)
+        self.g.proximal(self.tmp_dir + self.u, self.sigma, out = self.z)
```

Comment:
Can we move

`self.u += self.tmp_dir` to before this line and then pass `self.u`

The current code ends up allocating extra memory and seems to perform the operation twice.